src/app/games/apagos/ApagosRules.ts

```diff
@@ -0,0 +1,84 @@
+import { LegalityStatus } from 'src/app/jscaip/LegalityStatus';
+import { MGPNode } from 'src/app/jscaip/MGPNode';
+import { Player } from 'src/app/jscaip/Player';
+import { GameStatus, Rules } from 'src/app/jscaip/Rules';
+import { MGPMap } from 'src/app/utils/MGPMap';
+import { ApagosCoord } from './ApagosCoord';
+import { ApagosMessage } from './ApagosMessage';
+import { ApagosMove } from './ApagosMove';
+import { ApagosSquare } from './ApagosSquare';
+import { ApagosState } from './ApagosState';
+
+export class ApagosNode extends MGPNode<Rules<ApagosMove, ApagosState, LegalityStatus>, ApagosMove, ApagosState> {}
+
+export class ApagosRules extends Rules<ApagosMove, ApagosState> {
+
+    public static readonly singleton: ApagosRules = new ApagosRules(ApagosState);
+
+    public applyLegalMove(move: ApagosMove, state: ApagosState, status: LegalityStatus): ApagosState {
+        if (move.isDrop()) {
+            return this.applyLegalDrop(move, state);
+        } else {
+            return this.applyLegalSlideDown(move, state);
+        }
+    }
+    public applyLegalDrop(move: ApagosMove, state: ApagosState): ApagosState {
```

Comment:
should be private? Same for other variations of `applyLegalMove` and `isLegal`

Reply:
voilà!